WooCommerce/WooCommerceTests/ViewRelated/Orders/Issue Refund/IssueRefundViewModelTests.swift

```diff
@@ -228,11 +228,56 @@ final class IssueRefundViewModelTests: XCTestCase {
         let order = MockOrders().makeOrder(items: items)
 
         // When
-        let viewModel = IssueRefundViewModel(order: order, currencySettings: currencySettings)
+        let viewModel = IssueRefundViewModel(order: order, refunds: [], currencySettings: currencySettings)
         viewModel.selectAllOrderItems()
 
         // Then
         let selectedItemsTitle = String(format: NSLocalizedString("%d items selected", comment: ""), 6)
         XCTAssertEqual(viewModel.selectedItemsTitle, selectedItemsTitle)
     }
+
+    func test_viewModel_correcly_reduces_refunded_items() {
+        // Given
+        let currencySettings = CurrencySettings()
+        let items = [
+            MockOrderItem.sampleItem(itemID: 1, productID: 1, quantity: 3, price: 11.50),
+            MockOrderItem.sampleItem(itemID: 2, productID: 2, quantity: 2, price: 12.50),
+            MockOrderItem.sampleItem(itemID: 3, productID: 3, quantity: 1, price: 13.50),
+        ]
+        let order = MockOrders().makeOrder(items: items)
+        let refund = MockRefunds.sampleRefund(items: [
+            MockRefunds.sampleRefundItem(productID: 1, quantity: -2),
+            MockRefunds.sampleRefundItem(productID: 2, quantity: -1),
+        ])
+
+        // When
+        let viewModel = IssueRefundViewModel(order: order, refunds: [refund], currencySettings: currencySettings)
+
+        // Then
+        XCTAssertEqual(viewModel.quantityAvailableForRefundForItemAtIndex(0), 1)
+        XCTAssertEqual(viewModel.quantityAvailableForRefundForItemAtIndex(1), 1)
+        XCTAssertEqual(viewModel.quantityAvailableForRefundForItemAtIndex(2), 1)
+    }
+
+    func test_viewModel_correcly_filters_items_already_refunded() {
+        // Given
+        let currencySettings = CurrencySettings()
+        let items = [
+            MockOrderItem.sampleItem(itemID: 1, productID: 1, quantity: 3, price: 11.50),
+            MockOrderItem.sampleItem(itemID: 2, productID: 2, quantity: 2, price: 12.50),
+            MockOrderItem.sampleItem(itemID: 3, productID: 3, quantity: 1, price: 13.50),
+        ]
+        let order = MockOrders().makeOrder(items: items)
+        let refund = MockRefunds.sampleRefund(items: [
+            MockRefunds.sampleRefundItem(productID: 1, quantity: -3),
+            MockRefunds.sampleRefundItem(productID: 2, quantity: -2),
+        ])
+
+        // When
+        let viewModel = IssueRefundViewModel(order: order, refunds: [refund], currencySettings: currencySettings)
+
+        // Then
+        let itemRows = viewModel.sections[0].rows
+        XCTAssertEqual(itemRows.count, 2) // One item left + summary row
```

Comment:
just wondering, does it make sense to assert on the quantity like the previous test case `XCTAssertEqual(viewModel.quantityAvailableForRefundForItemAtIndex(0), 1)`?